restore-sql-server-transaction-logs/README.md

```diff
@@ -0,0 +1,309 @@
+
+        gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+        --member="serviceAccount:cloud-function-sql-restore-log@${PROJECT_ID}.iam.gserviceaccount.com" \
+        --role="projects/${PROJECT_ID}/roles/cloud.sql.importer"
+
+1. Deploy the cloud function that triggers when a new object is uploaded to the bucket. The function will restore the full and transaction log backup files and also handle the file synchronization on the bucket.
+
+- On your local development environment, install and initialize the gcloud CLI.
+
+- Clone the sql server restore cloud function repository.
+
+- Navigate to the restore-sql-server-transaction-logs/Function folder
+
+- From the restore-sql-server-transaction-logs/Function folder, run the following gcloud command to deploy the cloud function:
+
+        gcloud functions deploy <YOUR_FUNCTION_NAME> \
+        --gen2 \
+        --region=<YOUR_REGION> \
+        --retry \
+        --runtime=<YOUR_RUNTIME> \
+        --source=<YOUR_SOURCE_LOCATION> \
+        --entry-point=<YOUR_CODE_ENTRYPOINT> \        
+        --set-env-vars USE_FIXED_FILE_NAME_FORMAT=False,PROCESSED_BUCKET_NAME=,MAX_OPERATION_FETCH_TIME_SECONDS=30
+        --service-account cloud-function-sql-restore-log@${PROJECT_ID}.iam.gserviceaccount.com
+
+1. To invoke an authenticated cloud function, the underlying principal must have the invoker IAM permission. Assign the Invoker role (roles/run.invoker) through Cloud Run for 2nd gen functions to the function’s service account:
+
+        gcloud functions add-invoker-policy-binding <YOUR_FUNCTION_NAME> \
+        --region="<YOUR_REGION>" \
+        --member="serviceAccount:cloud-function-sql-restore-log@${PROJECT_ID}.iam.gserviceaccount.com"
+
+
+## Setup and configuration - Upload script
+
+
+For the operations in this section below you need the following permissions:
+
+                iam.serviceAccounts.create
+                iam.serviceAccounts.get
+                iam.serviceAccounts.list
+                iam.serviceAccountKeys.create
+                iam.serviceAccountKeys.delete
+                iam.serviceAccountKeys.disable
+                iam.serviceAccountKeys.enable
+                iam.serviceAccountKeys.get
+                iam.serviceAccountKeys.list
+                resourcemanager.projects.get
+                resourcemanager.projects.list
+                resourcemanager.projects.getIamPolicy
+                resourcemanager.projects.setIamPolicy
+
+or, your user must be member of the following predefined roles:
+
+                roles/iam.serviceAccountCreator
+                roles/iam.serviceAccountKeyAdmin
+                roles/resourcemanager.projectIamAdmin                
+
+Powershell is the language of choice for the upload script because many SQL Server Database Administrators (DBAs) leverage PowerShell as a valuable tool to streamline and automate their tasks and efficiently manage database environments. PowerShell runs on Windows, Linux, and macOS.

```

Comment:
What are the requirements to run these commands? (gcloud, powershell), and their versions?

Reply:
Done.

Reply:
Done.

Reply:
LGTM